pytensor/tensor/extra_ops.py

```diff
@@ -1583,27 +1595,294 @@ def broadcast_shape_iter(
+    """
+    Return numbers spaced evenly on a log scale (a geometric progression).
+
+    Parameters
+    ----------
+    Returns `num` evenly spaced samples, calculated over the interval [`start`, `stop`].
+
+    The endpoint of the interval can optionally be excluded.
+
+    Parameters
+    ----------
+    start: int, float, or TensorVariable
+        The starting value of the sequence.
+
+    stop: int, float or TensorVariable
+        The end value of the sequence, unless `endpoint` is set to False.
+        In that case, the sequence consists of all but the last of `num + 1` evenly spaced samples, such that `stop` is excluded.
+
+    num: int
+        Number of samples to generate. Must be non-negative.
+
+    base: float
+        The base of the log space. The step size between the elements in ln(samples) / ln(base)
+        (or log_base(samples)) is uniform.
+
+    endpoint: bool
+        Whether to include the endpoint in the range.
+
+    dtype: str, optional
+        dtype of the output tensor(s). If None, the dtype is inferred from that of the values provided to the `start`
+        and `end` arguments.
+
+    axis: int
+        Axis along which to generate samples. Ignored if both `start` and `end` have dimension 0. By default, axis=0
+        will insert the samples on a new left-most dimension. To insert samples on a right-most dimension, use axis=-1.
+
+    end:  int, float or TensorVariable
+        .. warning::
+            The "end" parameter is deprecated and will be removed in a future version. Use "stop" instead.
+        The end value of the sequence, unless `endpoint` is set to False.
+        In that case, the sequence consists of all but the last of `num + 1` evenly spaced samples, such that `end` is
+        excluded.
+
+    steps: float, int, or TensorVariable
+        .. warning::
+            The "steps" parameter is deprecated and will be removed in a future version. Use "num" instead.
+
+        Number of samples to generate. Must be non-negative
+
+    Returns
+    -------
+    samples: TensorVariable
+        Tensor containing `num` evenly-spaced values between [start, stop]. The range is inclusive if `endpoint` is True.
+    """
+    stop, num = _check_deprecated_inputs(stop, end, num, steps)
+    (start, stop), dtype = _broadcast_inputs_and_dtypes(start, stop, dtype=dtype)
+    start, stop, base = _broadcast_base_with_inputs(start, stop, base, dtype, axis)
+
+    out_sign = sign(start)
+    log_start, log_stop = (
+        log(start * out_sign) / log(base),
+        log(stop * out_sign) / log(base),
+    )
+    result = _linspace_core(
+        start=log_start,
+        stop=log_stop,
+        num=num,
+        endpoint=endpoint,
+        dtype=dtype,
+        axis=0,
+        retstep=False,
+    )
+    result = base**result
+
+    if num > 0:
+        set_subtensor(result[0, ...], start, inplace=True)
+        if num > 1 and endpoint:
+            set_subtensor(result[-1, ...], stop, inplace=True)
```

Comment:
Don't set inplace, that will be done in the rewrites.

Anyway we can allow `num` to be symbolic? If not, raise a `NotImplementedError`?

Reply:
I changed this to a switch with a python `if endpoint`, same as the other place you flagged. I think it should cover symbolic `num`, but I'll also add a fully symbolic test just to be sure.